src/main/java/org/folio/inventory/consortium/handlers/MarcInstanceSharingHandlerImpl.java

```diff
@@ -74,6 +93,63 @@ public Future<String> publishInstance(Instance instance, SharingInstance sharing
       });
   }
 
+  private Future<Record> unlinkAuthorityLinksIfNeeded(Record marcRecord, String instanceId, Context context, Storage storage) {
+    return entitiesLinksService.getInstanceAuthorityLinks(context, instanceId)
+      .compose(entityLinks -> {
+        if (entityLinks.isEmpty()) {
+          LOGGER.debug("unlinkAuthorityLinksIfNeeded:: Not found linked authorities for instance id: {} and tenant: {}", instanceId, context.getTenantId());
+          return Future.succeededFuture(marcRecord);
+        }
+        AuthorityRecordCollection authorityRecordCollection = storage.getAuthorityRecordCollection(context);
+        return entitiesLinksService.getLinkingRules(context)
+          .compose(linkingRules -> unlinkLocalAuthorities(entityLinks, linkingRules, marcRecord, instanceId, context, authorityRecordCollection));
+      });
+  }
+
+  private Future<Record> unlinkLocalAuthorities(List<Link> entityLinks, List<LinkingRuleDto> linkingRules, Record marcRecord,
+                                                String instanceId,Context context, AuthorityRecordCollection authorityRecordCollection) {
+
+    return getLocalAuthoritiesIdsList(entityLinks, authorityRecordCollection)
+      .compose(localAuthoritiesIds -> {
+        if (localAuthoritiesIds.isEmpty()) {
+          return Future.succeededFuture(marcRecord);
+        }
+        List<String> fields = linkingRules.stream().map(LinkingRuleDto::getBibField).toList();
+        LOGGER.debug("unlinkLocalAuthorities:: Unlinking from tenant: {} local authorities: {}", context.getTenantId(), localAuthoritiesIds);
+        if (MarcRecordUtil.removeSubfieldsThatContainsValues(marcRecord, fields, '9', localAuthoritiesIds)) {
```

Comment:
Please add comments to this if block. I am not sure why shared authority links are in play.

Reply:
Added a comment